File: examples/python/src/hello/about.py
"""CLI About slice: version + donate URL + update stub (no crash payload)."""

from __future__ import annotations

from importlib.metadata import PackageNotFoundError, version
from typing import TypedDict

DONATE_URL = "https://github.com/sponsors"


def _package_version() -> str:
    try:
        return version("golden-path-python")
    except PackageNotFoundError:
        return "0.1.0"


APP_VERSION = _package_version()


class AboutUpdate(TypedDict):
    status: str
    version: str | None
    url: str | None


class AboutPayload(TypedDict):
    version: str
    donate: str
    summary: str
    update: AboutUpdate


def about_summary() -> str:
    """Return a one-line About string."""
    return f"golden-path {APP_VERSION} donate {DONATE_URL}"


def about_payload() -> AboutPayload:
    """Return the shared About/donate/update JSON object."""
    return {
        "version": APP_VERSION,
        "donate": DONATE_URL,
        "summary": about_summary(),
        "update": {"status": "current", "version": None, "url": None},
    }

File: examples/rust/src/about.rs
//! CLI About slice: version + donate URL + update stub.

pub const APP_VERSION: &str = env!("CARGO_PKG_VERSION");
pub const DONATE_URL: &str = "https://github.com/sponsors";

pub fn summary() -> String {
    format!("golden-path {APP_VERSION} donate {DONATE_URL}")
}

/// Shared About/donate/update JSON (no serde — keep the stub dependency-free).
pub fn payload_json() -> String {
    format!(
        "{{\"version\":\"{APP_VERSION}\",\"donate\":\"{DONATE_URL}\",\"summary\":\"{}\",\"update\":{{\"status\":\"current\",\"version\":null,\"url\":null}}}}",
        summary()
    )
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn includes_version_and_donate() {
        let text = summary();
        assert!(text.contains(APP_VERSION));
        assert!(text.contains("donate"));
    }

    #[test]
    fn payload_json_matches_shared_contract() {
        let raw = payload_json();
        assert!(raw.contains(format!("\"version\":\"{APP_VERSION}\"").as_str()));
        assert!(raw.contains("\"status\":\"current\""));
        assert!(raw.contains("\"url\":null"));
        assert!(raw.contains(DONATE_URL));
    }
}
